diff --git a/autoload.module b/autoload.module
index aa5b9df..c048fa1 100644
--- a/autoload.module
+++ b/autoload.module
@@ -76,6 +76,20 @@ function autoload_form_alter(&$form, $form_state, $form_id) {
     case 'system_modules':
       $form['#submit'][] = 'autoload_form_submit_registry_rebuild';
       break;
+
+    case 'dblog_admin_settings':
+      $form['autoload'] = array(
+        '#type' => 'fieldset',
+        '#title' => t('Autoload'),
+        '#description' => t('Configure database logging for the autoload module.'),
+        '#weight' => -10,
+      );
+      $form['autoload']['autoload_dblog_enable'] = array(
+        '#type' => 'checkbox',
+        '#title' => t('Enable logging for autoload registry misses.'),
+        '#default_value' => variable_get('autoload_dblog_enable', TRUE),
+      );
+      break;
   }
 }
 
@@ -201,7 +215,7 @@ function _autoload_registry_check_code($type, $name = NULL) {
     require_once './' . $file;
     return TRUE;
   }
-  else {
+  elseif (variable_get('autoload_dblog_enable', TRUE)) {
     $variables = array(
       '%type' => $type,
       '%name' => $name,
